### PR TITLE
[REVIEW] FEA generic from_edgelist() and from_adjlist() APIs

### LINKED ISSUE #1141: [ENH] Remove from_cudf or form_dask_cudf references in methods
**Describe the solution you'd like**
Hi!

I have noticed that some methods contain the data type within the method name.

- from_cudf_adjlist(offset_col, index_col[, …])
- from_cudf_edgelist(input_df[, source, …]) 
- from_dask_cudf_edgelist(input_ddf[, source, …]) 

I'd propose a different to remove cudf and dask_cudf references in the method name, ending something like this:

- from_adjlist(offset_col, index_col[, …])
- from_edgelist(input_df[, source, …]) 
- from_edgelist(input_ddf[, source, …]) 

I think the data type could be inferred from the parameter passed, and that would allow keeping the method name while adding new data types support to those methods, like pandas, numpy, cupy, dask...

Hope it helps!

Regards,
Miguel

### PR DESCRIPTION
**Pull request by rlratzel**
Initial implementation and tests for generic `from_edgelist()` and `from_adjlist()` APIs, which can be used in place of `from_cudf_edgelist()` and `from_pandas_edgelist()`, etc. to make for more common code across different input data types.

closes #1141

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cugraph/structure/convert_matrix.py [hunk 1]
```diff
@@ -15,8 +15,62 @@
...
+    renumber : bool
+        If source and destination indices are not in range 0 to V where V
+        is number of vertices, renumber argument should be True.
+
+    Examples
+    --------
+    >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
+    >>>                   dtype=['int32', 'int32', 'float32'], header=None)
+    >>> G = cugraph.Graph()
+    >>> G = cugraph.from_edgelist(M, source='0', target='1', weight='2')
```
Comment (Iroy30) [inline python/cugraph/structure/convert_matrix.py:59]: weight ==> edge_attr?

Comment (rlratzel) [reply python/cugraph/structure/convert_matrix.py]: Thanks, fixed in the latest commit.

**Comment by BradReesWork** [comment]
rerun tests

**Review by BradReesWork — APPROVED** [review]

**Review by afender — APPROVED** [review]
Thanks. Initial naming came from consistency with 
https://networkx.org/documentation/stable/reference/generated/networkx.convert_matrix.from_pandas_edgelist.html

**Comment by rlratzel** [comment]
rerun tests

**Review by BradReesWork — APPROVED** [review]

**Pull request merged**